aws/src/test/java/org/apache/iceberg/aws/s3/MinioUtil.java

```diff
@@ -29,15 +29,18 @@
 import software.amazon.awssdk.services.s3.S3ClientBuilder;
 
 public class MinioUtil {
+  public static final String LATEST_TAG = "latest";
+  // This version doesn't support strong integrity checks
+  static final String LEGACY_TAG = "RELEASE.2024-12-18T13-15-44Z";
```

Comment:
can we try and roll back all of the changes in the PR and then only update the TAG to [RELEASE.2025-01-20T14-49-07Z](https://github.com/minio/minio/releases/tag/RELEASE.2025-01-20T14-49-07Z) (see also https://github.com/minio/minio/issues/20845#issuecomment-2604259537). I've tried it locally and that fixed the issue described in https://github.com/apache/iceberg/issues/12237 when running those tests locally with Docker Desktop

Reply:
@nastra it does, but it won't still work with some other S3-compatible storages like Vast, Dell ECS, so upgrading Minio to compatible version doesn't solve the issue

Reply:
@nastra We intentionally specified 2024-12 here. Using a newer tag hides the actual problem.